contracts/eosio.msig/include/eosio.msig/eosio.msig.hpp

```diff
@@ -123,52 +325,50 @@ namespace eosio {
          using invalidate_action = eosio::action_wrapper<"invalidate"_n, &multisig::invalidate>;
+   };
 
-      private:
-         struct [[eosio::table]] proposal {
-            name                            proposal_name;
-            std::vector<char>               packed_transaction;
-
-            uint64_t primary_key()const { return proposal_name.value; }
-         };
-
-         typedef eosio::multi_index< "proposal"_n, proposal > proposals;
+   struct [[eosio::table]] proposal {
+      name                                                            proposal_name;
+      std::vector<char>                                               packed_transaction;
+      eosio::eosio_msig_binary_extension< std::optional<time_point> > earliest_exec_time;
 
-         struct [[eosio::table]] old_approvals_info {
-            name                            proposal_name;
-            std::vector<permission_level>   requested_approvals;
-            std::vector<permission_level>   provided_approvals;
+      uint64_t primary_key()const { return proposal_name.value; }
+   };
+   typedef eosio::multi_index< "proposal"_n, proposal > proposals;
 
-            uint64_t primary_key()const { return proposal_name.value; }
-         };
-         typedef eosio::multi_index< "approvals"_n, old_approvals_info > old_approvals;
+   struct [[eosio::table]] old_approvals_info {
+      name                            proposal_name;
+      std::vector<permission_level>   requested_approvals;
+      std::vector<permission_level>   provided_approvals;
 
-         struct approval {
-            permission_level level;
-            time_point       time;
-         };
+      uint64_t primary_key()const { return proposal_name.value; }
+   };
+   typedef eosio::multi_index< "approvals"_n, old_approvals_info > old_approvals;
 
-         struct [[eosio::table]] approvals_info {
-            uint8_t                 version = 1;
-            name                    proposal_name;
-            //requested approval doesn't need to cointain time, but we want requested approval
-            //to be of exact the same size ad provided approval, in this case approve/unapprove
-            //doesn't change serialized data size. So, we use the same type.
-            std::vector<approval>   requested_approvals;
-            std::vector<approval>   provided_approvals;
+   struct approval {
+      permission_level level;
+      time_point       time;
+   };
 
-            uint64_t primary_key()const { return proposal_name.value; }
-         };
-         typedef eosio::multi_index< "approvals2"_n, approvals_info > approvals;
+   struct [[eosio::table]] approvals_info {
+      uint8_t                 version = 1;
+      name                    proposal_name;
+      //requested approval doesn't need to cointain time, but we want requested approval
+      //to be of exact the same size ad provided approval, in this case approve/unapprove
```

Comment:
"to be of exact the same size ad provided" -> "to be of exactly the same size as provided"

Reply:
Updated comment.